client/src/pages/Register/RegisterStep.tsx

```diff
@@ -1,67 +1,211 @@
 import React from 'react';
+import debounce from 'lodash/debounce';
+import { useHistory } from 'react-router-dom';
 
+import debug from 'utils/debug';
+import Spinner from 'components/Spinner';
 import { type ErrorTransformer } from '@rjsf/utils';
-
-import JsonSchemaForm from 'components/JsonSchemaForm';
+import ErrorBoundary from 'components/ErrorBoundary';
+import JsonSchemaForm, {
+  calculatePrice,
+  FormData,
+  JsonSchemaFormChangeEvent,
+}from 'components/JsonSchemaForm';
 import Template from 'components/Template';
+import api from 'store/register/api';
+import {
+  useRegFormData,
+  setRegFormData,
+  setTotals,
+  setPaymentStep,
+  useAppDispatch,
+} from 'store/register/store';
 
-import type { RegisterStepProps } from './component';
-import RegistrationClosed from './RegistrationClosed';
+import PageWrapper from './PageWrapper';
 import transformErrorsCreator from './transformErrors';
 
-function RegistrationStep(props: RegisterStepProps) {
-  const [liveValidate, setLiveValidate] = React.useState(false);
+import './styles.scss';
 
-  if (props.step !== 'registration') return null;
+const getLocalStorageKey = (config: ApiRegister) => {
+  const name = config.dataSchema.title || 'formData';
+  const date = config.event.start;
 
-  if (!props.config.event.is_open && !props.config.invitation) {
-    return <RegistrationClosed {...props} />;
+  return `${name}, ${date.year}-${date.month}-${date.day}`;
+}
+
+const saveFormDataToLocalStorage = debounce((formData, config) => {
```

Comment:
I decided to so a simple DIY version of "redux persist" instead of adding the whole package